src/txn/txn_recover.c

```diff
@@ -996,6 +999,17 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
     if (eviction_started)
         WT_TRET(__wt_evict_destroy(session));
 
+    /*
```

Comment:
Is this just to make sure the new write gen gets written to the disk? Is there a reason we have to close dhandles as opposed to just marking the tree as `modified`?

Reply:
After chatting with @raviprakashgiri29 I think I understand.

@kommiharibabu 
Just confirming my understanding. My thought was that wiping the transaction ids happens lazily whenever we read an onpage value so the only thing we'd have to do in that case is update the `base_write_gen` and there'd be no need to flush everything out of the cache like this. Is this necessary because we may have updates in the chain that have transaction ids?

Reply:
@tetsuo-cpp Yes, it is possible that the pages in the cache may have transaction ids when they are loaded into the cache as part of rollback to stable. Until the pages are flushed and read again, those transaction ids never get reset. We want to achieve the reset by closing the handles. An alternative way to discard all the pages in the cache and increment all the btree specific `write_gen` instead of closing handles.

Reply:
@kommiharibabu 
Yeah, it's specifically stuff in the update chain. Not the on-disk values. This is the part that I was confused with.